javascript/ql/src/semmle/javascript/frameworks/Koa.qll

```diff
@@ -100,6 +122,77 @@ module Koa {
     }
   }
 
+  /**
+   * A Koa route handler registered using a routing library.
+   *
+   * Example of what that could look like:
+   * ```JavaScript
+   * const router = require('koa-router')();
+   * const Koa = require('koa');
+   * const app = new Koa();
+   * router.get('/', async (ctx, next) => {
+   *    // route handler stuff
+   * });
+   * app.use(router.routes());
+   * ```
+   */
+  private class RoutedRouteHandler extends RouteHandler, DataFlow::SourceNode {
+    DataFlow::InvokeNode router;
+    DataFlow::MethodCallNode call;
+
+    RoutedRouteHandler() {
+      router = DataFlow::moduleImport(["@koa/router", "koa-router"]).getAnInvocation() and
+      call = router.getAMethodCall*() and
+      call.getMethodName() =
+        [
+          "use", "get", "post", "put", "link", "unlink", "delete", "del", "head", "options",
+          "patch", "all"
+        ] and
+      this.flowsTo(call.getArgument(any(int i | i >= 1)))
+    }
+
+    override DataFlow::SourceNode getRouteHandlerRegistration() {
+      result = call
+      or
+      result = router.getAMethodCall("routes")
```

Comment:
Does this model the mentioned `app.use(router.routes())` bit above?
Shouldn't the registration be the `app.use(...)` call instead then? My intuition is that `router.routes()` simply is a getter.